dashboard/modules/serve/serve_head.py

```diff
@@ -37,32 +52,103 @@ async def get_serve_instance_details(self, req: Request) -> Response:
             )
 
         try:
-            async with self._http_session.get(
-                f"{head_agent_address}/api/serve/applications/"
+            req_data = await req.read()
+            headers = {"content-type": req.content_type}
+            rest_method_executor = self.rest_method_executors[method]
+            url = f"{head_agent_address}{route}"
+            async with rest_method_executor(
+                url=url, data=req_data, headers=headers
             ) as resp:
-                if resp.status == 200:
-                    result_text = await resp.text()
-                    return Response(
-                        text=result_text,
-                        content_type="application/json",
-                    )
-                else:
-                    status = resp.status
-                    error_text = await resp.text()
-                    raise Response(
-                        status=500,
-                        text=f"Request failed with status code {status}: {error_text}.",
-                    )
-        except Exception as e:
+                resp_text = await resp.text()
+                return Response(
+                    text=resp_text,
+                    status=resp.status,
+                    content_type=resp.content_type,
+                )
+        except Exception:
+            import traceback
```

Comment:
why delayed import?

Reply:
Ah, I did that when I was experimenting. I moved it to the top of the file, thanks for the catch!